### ISSUE TITLE
🐛 Biome show "This let declares a variable that is only assigned once." when using let keyword to define Svelte 5 component's state

### ISSUE THREAD

**Issue by x4fingers** [biomejs/biome#4484]
### Environment information

```block
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.14.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "bun/1.1.31"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

<img width="398" alt="image" src="https://github.com/user-attachments/assets/d2dc7dbb-2f12-4da2-8fa4-2063ab619a0d">

This error is shown at the 4th line where I define a Svelte 5 component state.

Please check my reproduction repository here: https://github.com/x4fingers/biome-repro-1730968820894

### Expected result

It should not throw the error because a Svelte 5 component's state needs to be declared with `let` keyword to be bindable on input.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
This is a known and documented limitation: https://biomejs.dev/internals/language-support/#html-super-languages-support

You'll have to disable the rule for svelte files.

**Comment by x4fingers** [comment]
@ematipico can you explain why it's not planned instead of being the default configuration for Svelte?

**Comment by ematipico** [comment]
Sure. At the moment Biome has partial support, which comes with some trade-offs and many workarounds to make sure we provide a **decent enough** DX to the users.

In order to provide a default configuration, we need to add more workarounds, which will cause possible breaking changes to the users once we will achieve full support. Plus, it adds more maintenance burden to the big code base, risking overlooking it when landing full support.

However, with lint rules, users can turn them on and off at their discretion, and once we land full support, users would need to remove the override whenever they want. 

Between breaking changes and user configuration, we decided to go with the configuration, because we can also add a migration path (automatic), and users can update it at their leisure.
